pkg/controller/scyllacluster/sync_services.go

```diff
@@ -0,0 +1,303 @@
+package scyllacluster
+
+import (
+	"context"
+	"fmt"
+	"regexp"
+	"strconv"
+
+	scyllav1 "github.com/scylladb/scylla-operator/pkg/api/scylla/v1"
+	"github.com/scylladb/scylla-operator/pkg/controller/helpers"
+	"github.com/scylladb/scylla-operator/pkg/controller/scyllacluster/resource"
+	"github.com/scylladb/scylla-operator/pkg/naming"
+	"github.com/scylladb/scylla-operator/pkg/resourceapply"
+	appsv1 "k8s.io/api/apps/v1"
+	corev1 "k8s.io/api/core/v1"
+	policyv1beta1 "k8s.io/api/policy/v1beta1"
+	apierrors "k8s.io/apimachinery/pkg/api/errors"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	utilerrors "k8s.io/apimachinery/pkg/util/errors"
+	"k8s.io/klog/v2"
+)
+
+var serviceOrdinalRegex = regexp.MustCompile("^.*-([0-9]+)$")
+
+func (scc *Controller) makeServices(sc *scyllav1.ScyllaCluster, oldServices map[string]*corev1.Service) []*corev1.Service {
+	services := []*corev1.Service{
+		resource.HeadlessServiceForCluster(sc),
+	}
+
+	for _, rack := range sc.Spec.Datacenter.Racks {
+		stsName := naming.StatefulSetNameForRack(rack, sc)
+
+		for ord := int32(0); ord < rack.Members; ord++ {
+			svcName := fmt.Sprintf("%s-%d", stsName, ord)
+			oldSvc := oldServices[svcName]
+			services = append(services, resource.MemberService(sc, rack.Name, svcName, oldSvc))
+		}
+	}
+
+	return services
+}
+
+func (scc *Controller) syncServices(
```

Comment:
yeah, I've tried to separate each block with comments to retain readability. The issue is that if you start extracting functions from the business logic the returns there are gonna introduce another return variable that needs to be checked on the call site to be propagated.

Reply:
not necessarly. For example loop which deletes services could be extracted without any additional return value. It doesn't have any early non-error return. Extracting it would reduce LOC by ~50. 
Extracting podEviction code would save another ~20.

Reply:
done

Reply:
done , except the eviction, the smallest operation there is deleting a node which includes deleting the service which requires handling requeues